pkg/apis/v1beta1/nodepool.go

```diff
@@ -202,3 +210,72 @@ func (pl *NodePoolList) OrderByWeight() {
 		return ptr.Int32Value(pl.Items[a].Spec.Weight) > ptr.Int32Value(pl.Items[b].Spec.Weight)
 	})
 }
+
+// GetAllowedDisruptions returns the minimum allowed disruptions across all disruption budgets for a given node pool.
+// This returns two values as the resolved value for a percent depends on the number of current node claims.
+func (in *NodePool) GetAllowedDisruptions(ctx context.Context, c clock.Clock) (intstr.IntOrString, intstr.IntOrString, error) {
```

Comment:
I wasn't quite getting this from the description, but my assumption is that one of these values is going to be a percentage value and the other is going to be an integer value. Is there a reason that they are both `IntOrString` since one should always be a string and the other should always be an integer?

Reply:
The IntOrString has a Type value that allows the GetScaledValue to infer whether it needs to do a percentage or a raw value. I could probably put both of the values in the return value here, but we'd need to in-place modify the type to compare the two resolved values together, where I think having two values may just more deliberate here.